core/src/main/scala/io/snappydata/impl/ComplexTypeSerializerImpl.scala

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2016 SnappyData, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+package io.snappydata.impl
+
+import java.util
+
+import com.pivotal.gemfirexd.snappy.ComplexTypeSerializer
+
+import org.apache.spark.sql.catalyst.expressions.GenericRowWithSchema
+import org.apache.spark.sql.catalyst.expressions.codegen.BufferHolder
+import org.apache.spark.sql.store.CodeGeneration
+import org.apache.spark.sql.types.{DataType, StructType}
+
+/**
+ * Implementation of <code>ComplexTypeSerializer</code>.
+ */
+final class ComplexTypeSerializerImpl(dataType: DataType)
+    extends ComplexTypeSerializer {
+
+  def this(tp: java.lang.Object) = this(SparkShellRDDHelper.getDataType(tp))
+
+  private[this] val (serializer, converter) = CodeGeneration
+      .getComplexTypeSerializer(dataType)
+
+  private[this] val struct = dataType match {
+    case s: StructType => Some(s)
+    case _ => None
+  }
+
+  private[this] val bufferHolder = new BufferHolder()
```

Comment:
The CodeGeneration.getComplexTypeSerializer is one per JVM. The ComplexTypeSerializer can be created as user sees fit for each type being used. Normal way would be to create one per type in code-fragment (e.g. a method, "session" etc) of user code.